### DISCUSSION TITLE
The Edge Function size exceeds 1MB

### DISCUSSION THREAD

**Discussion by vignesh-gupta** [vercel/next.js#74441] [Help] [answered]
### Summary

Using hono with nextjs from [this doc](https://hono.dev/docs/getting-started/vercel),  Started getting the size limit exceeded after [this commit](https://github.com/vignesh-gupta/clikz/commit/968609f6caa105a604dee11470762b52bb05deea). Which has nothing to do with api route. 

### Additional information

```js
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Home Single Language
  Available memory (MB): 7540
  Available CPU cores: 12
Binaries:
  Node: 20.18.0
  npm: 10.9.0
  Yarn: 1.22.22
  pnpm: 9.14.2
Relevant Packages:
  next: 15.1.3 // Latest available version is detected (15.1.3).
  eslint-config-next: N/A
  react: 19.0.0
  react-dom: 19.0.0
  typescript: N/A
Next.js Config:
  output: N/A
```

### Example

https://github.com/vignesh-gupta/clikz/

**Comment by icyJoseph** [comment, marked as answer]
Could you make a build with, https://www.npmjs.com/package/@next/bundle-analyzer activated, and see what the Edge Report (three reports open, one for your Node env, one for the browser, and one for the Edge), is showing as included?

**Reply by icyJoseph** [reply]
I think the issue is somehow here... but how is it getting triggered....

<img width="1478" alt="Screenshot 2024-12-31 at 19 17 42" src="https://github.com/user-attachments/assets/0df76596-30a4-46e4-9cbb-e3b926535fdd" />

<img width="327" alt="Screenshot 2024-12-31 at 19 18 09" src="https://github.com/user-attachments/assets/608cde87-b011-47e4-93b1-18ab6b5769a4" />

**Reply by icyJoseph** [reply]
If I checkout, https://github.com/vignesh-gupta/clikz/commit/04030be90ea5a054a3d2caadbbea677b51c71381, parent of the https://github.com/vignesh-gupta/clikz/commit/968609f6caa105a604dee11470762b52bb05deea commit, then I see this breakdown:

<img width="1462" alt="Screenshot 2024-12-31 at 19 22 35" src="https://github.com/user-attachments/assets/f690d47e-ed77-49fc-b74c-32bce4b15486" />

**Reply by vignesh-gupta** [reply]
@icyJoseph thanks for the insight, I'll try to look in to this. Also I've tried on my side same, but here's a confusion I see -
As per the doc the max size limit of 1 MB is after after gzip compression. 
![image](https://github.com/user-attachments/assets/d999c20f-95e3-4020-a137-0c0cae634bc1)


And as I see in the analyzer, we are looking as `Parsed` sizes and note `Gzipped`. Here's `Gzipped` snapshot (which is less than 1 MB)
![image](https://github.com/user-attachments/assets/d920e832-861b-456b-aa4b-f9ce69057c8a)

**Reply by icyJoseph** [reply]
That's true, I overlooked that. Though I wonder what exactly is being accounted for. This is a Vercel platform issue though. 

From our part we can try to figure out why the edge bundle is larger anyway. Homo for sure, but is there anything else? Mixed imports/exports? I couldn't see any, though.

**Reply by vignesh-gupta** [reply]
From what I see, the both cases there are 2 major packages
1. Next, react, react-dom
2. prisma


Also I'll create a issue on vercel platform too for `gzip` size issue

**Reply by icyJoseph** [reply]
I wonder, if you could use, https://nextjs.org/docs/app/api-reference/config/next-config-js/output#caveats outputFileTracingExcludes on your route, to drop some unused stuff

**Reply by vignesh-gupta** [reply]
@icyJoseph also how can I raise the issue for vercel team to take a look?

And I've hono with next js in another project too, which has more endpoint but works fine. 

I managed to remove Prisma from `middleware` and reduce the size but still it's .09 MB extra. And didn't find any issues with hono either for import export.

![image](https://github.com/user-attachments/assets/5dec9be8-13bf-4e61-aeb3-33c23072c42e)

**Reply by icyJoseph** [reply]
I think you can either submit a support ticket, or go to the Vercel cli repository, called Vercel.

Good that you narrowed down the size though.

**Reply by icyJoseph** [reply]
btw, may I ask why do you need a manual stable sort? AFAIK, V8, introduced stable sort back in 2019: https://v8.dev/features/stable-sort

Is there other engines you know of, which do not have stable sort? Would like to know more.

**Reply by vignesh-gupta** [reply]
Hey @icyJoseph, apologies I missed the notification of this discussion. No I'm not with very deep understanding with that, but that stable sort is just name I remembered honestly. I'm only using to sort all the listeners on the priority basis.

Also I did figured a way out by avoiding edge function and use node runtime instead in hono. So thanks alot, I got to know about analyzing the package!

**Reply by vignesh-gupta** [reply]
Also I've mark as answered for other in case someone faces this issue so they can analyze their bundle too,
